crates/wasmtime/src/component/types.rs

```diff
@@ -0,0 +1,1025 @@
+use crate::component::func::{self, Lift, Memory, Options};
+use crate::component::values::{
+    self, Enum, Expected, Flags, List, Option, Record, Tuple, Union, Val, Variant,
+};
+use crate::store::StoreOpaque;
+use crate::ValRaw;
+use anyhow::{anyhow, bail, Context, Error, Result};
+use std::collections::HashMap;
+use std::fmt;
+use std::iter;
+use std::ops::Deref;
+use std::sync::Arc;
+use wasmtime_component_util::{DiscriminantSize, FlagsSize};
+use wasmtime_environ::component::{
+    ComponentTypes, InterfaceType, TypeEnumIndex, TypeExpectedIndex, TypeFlagsIndex,
+    TypeInterfaceIndex, TypeRecordIndex, TypeTupleIndex, TypeUnionIndex, TypeVariantIndex,
+};
+
+#[derive(Clone)]
+pub struct Handle<T> {
```

Comment:
This is a bit bike-sheddy, but instead of 

```rust
pub struct Handle<T> { ... }
pub struct TupleIndex(TypeTupleIndex);
pub enum Type {
    // ...
    Tuple(Handle<TupleIndex>),
    // ...
}
```

could this instead be:

```rust
struct Handle<T> { ... }
pub struct Tuple(Handle<TypeTupleIndex>);
pub enum Type {
    // ...
    Tuple(Tuple),
    // ...
}
```

That I think will allow us to be a bit more flexible in the future with internal representations ideally.

Reply:
I think this organization also helps method discovery since today there's lots of impls for `Handle<T>` for different `T` but for `Tuple` there will be just the methods related to tuples.